src/datasets/arrow_dataset.py

```diff
@@ -3738,33 +3727,33 @@ def _select_with_indices_mapping(
     @fingerprint_transform(inplace=False, ignore_kwargs=["load_from_cache_file", "indices_cache_file_name"])
     def sort(
         self,
-        column: str,
-        reverse: bool = False,
-        kind: str = None,
-        null_placement: str = "last",
+        column_names: Union[str, Sequence_[str]],
+        reverse: Union[bool, Sequence_[bool]] = False,
+        null_placement: str = "at_end",
         keep_in_memory: bool = False,
         load_from_cache_file: bool = True,
         indices_cache_file_name: Optional[str] = None,
         writer_batch_size: Optional[int] = 1000,
         new_fingerprint: Optional[str] = None,
+        kind="deprecated",
     ) -> "Dataset":
-        """Create a new dataset sorted according to a column.
+        """Create a new dataset sorted according to a single or multiple columns.
+        The transformation is applied to all the datasets of the dataset dictionary.
 
-        Currently sorting according to a column name uses pandas sorting algorithm under the hood.
-        The column should thus be a pandas compatible type (in particular not a nested type).
+        Currently sorting according to a column name(s) uses pyarrow's sort_indices sorting algorithm under the hood.
+        The column(s) should thus be a pyarrow compatible type (in particular not a nested type).
         This also means that the column used for sorting is fully loaded in memory (which should be fine in most cases).
 
         Args:
-            column (`str`):
-                Column name to sort by.
-            reverse (`bool`, defaults to `False`):
-                If `True`, sort by descending order rather then ascending.
-            kind (`str`, *optional*):
-                Pandas algorithm for sorting selected in `{quicksort, mergesort, heapsort, stable}`,
-                The default is `quicksort`. Note that both `stable` and `mergesort` use `timsort` under the covers and, in general,
-                the actual implementation will vary with data type. The `mergesort` option is retained for backwards compatibility.
-            null_placement (`str`, defaults to `last`):
-                Put `None` values at the beginning if first; last puts `None` values at the end.
+            column_names (`Union[str, Sequence[str]]`):
+                Either a single column name or a list of unqiue column names to sort by.
+            reverse (`Union[bool, Sequence[bool]]`, defaults to `False`):
+                If `True`, sort by descending order rather than ascending. If a single bool is provided,
+                the value is applied to the sorting of all column names. Otherwise a list of bools with the
+                same length and order as column_names must be provided.
+            null_placement (`str`, defaults to `at_end`):
+                Put `None` values at the beginning if `at_start` or `first`; `at_end` or `last` puts `None`
```

Comment:
```suggestion
                Put `None` values at the beginning if `at_start` or `first` or at the end  if `at_end` or `last` 
```

Reply:
Addressed separately in https://github.com/huggingface/datasets/pull/5502/commits/53c0a537b5c5c55f5a2aa72715b39175cc690bf4 because of minor typo